docs/contribute/plugin/plugin_requirement.md

```diff
@@ -0,0 +1,336 @@
+
+- **Rust:** To develop WasmEdge plugins using Rust, developers need to have Rust installed on their system. Instructions for installing Rust can be found in the [e Rust Setup Guide](/develop/rust/setup.md). Once Rust is installed, developers can start developing WasmEdge plugins using the Rust programming language.
+
+The choice of programming language depends on the developer's familiarity and skill with the language. C and C++ are commonly chosen for their low-level control over system resources and the ability to create high-performance plugins. Rust, on the other hand, is known for its safety features and performance.
+
+> It is recommended that developers choose the WasmEdge [C API](develop_plugin_c.md) for plugin development.
+
+## Set up the development environment
+
+To start developing WasmEdge plugins, it is essential to set up the development environment properly. This section provides step-by-step instructions for WasmEdge plugin development -
+
+- **Install a WasmEdge runtime**: You can download the latest version of WasmEdge from [GitHub repository](https://github.com/wasmEdge/wasmEdge). Here are the installation instructions for different operating system for [installing WasmEdge](/develop/build-and-run/install.md).
+  
+- **Install WasmEdge with plugins (optional)**: Installing WasmEdge with existing plugins can provide you with additional functionality and serve as a reference for your own plugin development. If you want to utilize or test the compatibility of your new plugin with existing plugins, you can install them using the provided installer script. The installed plugins will be available for your development environment.
+
+  To see a list of supported plugins and their specific install commands, see the [Install WasmEdge plugins and dependencies](develop/build-and-run/install/#install-wasmedge-plugins-and-dependencies) section.
+
+- **Install a compatible compiler**: The following compilers can be used to compile C and C++ code into WebAssembly bytecode:
+
+  - LLVM: `sudo apt-get install llvm`
+  - GCC: `sudo apt-get install gcc`
+
+  Rust has its own built-in compiler, which can be installed using rustup:
+
+  ```bash
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+  source $HOME/.cargo/env
+  ```
+
+- **Install necessary tools and dependencies**: You can use any text editor or IDE of your choice to write code. Here are some popular options:
+
+  - [Visual Studio Code](https://code.visualstudio.com/)
+  - [Atom](https://atom.io/)
+  - [Sublime Text](https://www.sublimetext.com/)
+
+  For debugging, you can use GDB, LLDB, or other debuggers that support WebAssembly. To install GDB, run `sudo apt-get install gdb`.
+
+- **Enable specific backends or additional components (if applicable):** Some plugins may require enabling specific backends or additional components to extend their functionality. The following links provide instructions for enabling specific backends in WasmEdge:
```

Comment:
Why do we mention Pytorch, TensorFlow, and OpenVINO here?

Reply:
Its an optional step. If someone plans to develop some plugin which is required Pytorch, TensorFlow, and OpenVINO backend they can install it following the links. 

Should I update it?